h/static/scripts/test/widget-controller-test.js

```diff
@@ -441,6 +444,27 @@ describe('WidgetController', function () {
     });
   });
 
+  describe('deferred websocket connection', function () {
+    it('should connect the websocket the first time the sidebar opens', function () {
+      $rootScope.$emit('sidebarOpened');
+      assert.called(fakeStreamer.connect);
+    });
+
+    describe('when logged in user changes', function () {
+      it('should not reconnect if the sidebar is closed', function () {
+        $rootScope.$emit(events.USER_CHANGED);
+        assert.notCalled(fakeStreamer.connect);
```

Comment:
This test _should_ be failing given the current logic in `widget-controller.js`, but isn't because this checks for a call to `connect` rather than `reconnect`.